app/controllers/evidence_controller.rb

```diff
@@ -140,6 +148,11 @@ def destroy
 
   private
 
+  def autogenerate_issue
+    @evidence.issue = Issue.autogenerate_from(@evidence)
```

Comment:
you only need `@evidence` to pass the node: in `Issue.create`, which is just to tie the Issue to the project, and the controller already has a reference to the project.

So from a caller/controller perspective `autogenerate_from(evidence)` makes sense, but from a receiver/model perspective, not so much. It would make more sense to pass a reference to the project. Perhaps `autogenerate_for(current_project)`? Or maybe there's a way to go through `current_project` so we skip the call to the issues_library node, something like:

```
current_project.issues.autogenerate!
```
if that sets the right node_id:, then we're golden

Reply:
We need the author too. We can also put it in `models/projects.rb` but it's weird.
```ruby
def autogenerate_issue_with_author(author)
  issues.create(
    author: author,
    node: issue_library
  )
end
```